.opencode/agents/type-design-analyzer.md

```
@@ -0,0 +1,72 @@
+- Base class with too many abstract methods (interface bloat)
+
+## Severity Levels
+
+- **HIGH**: Invalid states possible, invariants not enforced, primitive obsession
+- **MEDIUM**: Encapsulation leaks, suboptimal type choice, inheritance issues
+- **LOW**: Minor type design improvements
+
+## Report Format
+
+For each finding:
+```
```

markdownlint-cli2 (0.22.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)